src/http_protocol.cpp

```diff
@@ -31,6 +31,45 @@ std::string url_decode(const std::string& encoded) {
   return decoded_str;
 }
 
+bool is_method_allowed(uint8_t mask, method_t method) {
+  switch (method) {
+    case method_t::OPTIONS:
+      return (mask & OPTIONS_BITMASK) != 0;
+    case method_t::GET:
+      return (mask & GET_BITMASK) != 0;
+    case method_t::HEAD:
+      return (mask & HEAD_BITMASK) != 0;
+    case method_t::POST:
+      return (mask & POST_BITMASK) != 0;
+    case method_t::PUT:
+      return (mask & PUT_BITMASK) != 0;
+    case method_t::DELETE:
+      return (mask & DELETE_BITMASK) != 0;
+    case method_t::TRACE:
+      return (mask & TRACE_BITMASK) != 0;
+    case method_t::CONNECT:
+      return (mask & CONNECT_BITMASK) != 0;
+    default:
+      return false;
+  }
+}
```

Comment:
this should be simple, you just have to make method_t an enum class 
```suggestion
bool is_method_allowed(uint8_t mask, method_t method) {
  return mask & method;
}
```